### PR TITLE
Type target concept

### PR DESCRIPTION
**Pull request by mlevieux**
Functions in the code take a `target` which is originally a string, which authorized values are among an enumeration. The user of the code has to know those values to use the SDK correctly and that is a negative point, as the SDK to be the most self-explaining possible.
This branch implements a new type `Target` on top of the builtin `string`. It has a `Hashed` method returning its hashed equivalent. A number of `const` of the possible valid value for a `Target` are exported so the user can take advantage of them directly.

### CODE REVIEW

**Comment by blastrock** [comment]
I think this is a good change, but it should be done in other languages too. At the moment, all identity SDKs have an equivalent API.

So, I don't know if we should change this, and maybe later change the other SDKs. Or not merge this and do it everywhere first. Maybe you can take another opinion from Max or Loic.

Anyway, do not forget to update the doc/guides if you merge this.

**Review by blastrock — APPROVED** [review]

**Comment by mlevieux** [comment]
Ping @maximerety  for this change. See @blastrock's comment above. What do you think?

**Pull request closed without merging**
